### PR TITLE
New serverless pattern - s3-lambda-cdk-dotnet

### PR DESCRIPTION
**Pull request by surbhitas**
The CDK stack creates an S3 bucket and AWS Lambda function and demonstrate S3 integration with lambda.

### CODE REVIEW

File: s3-lambda-cdk-dotnet/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# AWS Lambda function that reads file information when file is uploaded in Amazon S3 bucket using CDK.
+
+This AWS CDK application demonstrates triggering a AWS Lambda function, when a file is uploaded in a S3 bucket.
+
+This pattern provides a solution for accessing the file information from AWS Lambda function, when a new file is uploaded in S3 bucket. An ideal use case for this is reading various file types e.g. image, csv, text, pdf files and extracting its metadata information and other file contents for futher processing or just passing this information to downstream systems.
```
Comment (bfreiberg) [inline s3-lambda-cdk-dotnet/README.md:5]:
```suggestion
This pattern provides a solution for accessing the file information from AWS Lambda function, when a new file is uploaded to a S3 bucket. An ideal use case for this is reading various file types e.g. image, csv, text, pdf files and extracting its metadata information and other file contents for further processing or just passing this information to downstream systems.
```

File: s3-lambda-cdk-dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    dotnet build src
+    ```
+4. Deploy the stack to your default AWS account and region.
+    ```
+    cdk deploy
+    ```
+
+## Testing
+
+1. After deployment, upload a sample file in S3 bucket created from the stack.
```
Comment (bfreiberg) [inline s3-lambda-cdk-dotnet/README.md:43]:
```suggestion
1. After deployment, upload a sample file to the S3 bucket that was created by the stack.
```

File: s3-lambda-cdk-dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+2. Check the lambda logs in cloudwatch log groups. You should see the file name of file you uploaded, in the function logs.
```
Comment (bfreiberg) [inline s3-lambda-cdk-dotnet/README.md:44]:
```suggestion
2. Check the Lambda function logs in the CloudWatch log groups. You should see the file name of file you uploaded, in the function logs.
```

File: s3-lambda-cdk-dotnet/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "title": "AWS Lambda function that reads file information when file is uploaded in Amazon S3 bucket using CDK.",
+  "description": "This AWS CDK application demonstrates triggering a AWS Lambda function, when a file is uploaded in a S3 bucket.",
+  "language": ".NET",
+  "level": "200",
+  "framework": "CDK",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "This pattern provides a solution for accessing the file information from AWS Lambda function, when a new file is uploaded in S3 bucket. An ideal use case for this is reading various file types e.g. image, csv, text, pdf files and extracting its metadata information and other file contents for futher processing or just passing this information to downstream systems."
```
Comment (bfreiberg) [inline s3-lambda-cdk-dotnet/example-pattern.json:10]:
```suggestion
      "This pattern provides a solution for accessing the file information from an AWS Lambda function, when a new file is uploaded in S3 bucket. An ideal use case for this is reading various file types e.g. image, csv, text, pdf files and extracting its metadata information and other file contents for further processing or just passing this information to downstream systems."
```

File: s3-lambda-cdk-dotnet/README.md (hunk 1 above)
Comment (surbhitas) [reply s3-lambda-cdk-dotnet/README.md]: resolved comments.

File: s3-lambda-cdk-dotnet/example-pattern.json (hunk 4 above)
Comment (surbhitas) [reply s3-lambda-cdk-dotnet/example-pattern.json]: resolved comments

File: s3-lambda-cdk-dotnet/src/lambda/src/lambda/Function.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        /// A simple function that takes a string and does a ToUpper
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public void FunctionHandler(S3Event evnt, ILambdaContext context)
+        {
+            foreach (var record in evnt.Records)
+            {
+                LambdaLogger.Log("----> File Name: " + record.S3.Object.Key + " Opration: " + record.EventName);
```
[the remark this answers is not in the document]
Comment (surbhitas) [reply s3-lambda-cdk-dotnet/src/lambda/src/lambda/Function.cs]: resolved comments

File: s3-lambda-cdk-dotnet/src/lambda/src/lambda/Readme.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# AWS Lambda Empty Function Project
```
[the remark this answers is not in the document]
Comment (surbhitas) [reply s3-lambda-cdk-dotnet/src/lambda/src/lambda/Readme.md]: removed the file.

**Comment by surbhitas** [comment]
Hi @bfreiberg,
I have addressed your comments. Can you review, please ?

**Pull request merged**